lib/nexpose/error.rb

```diff
@@ -15,9 +15,9 @@ def to_s
   end
 
   class AuthenticationFailed < APIError
-    def initialize(req)
+    def initialize(req, reason = nil)
       @req = req
-      @reason = "Login Failed"
+      @reason = reason ||= "Login Failed"
```

Comment:
`@reason = reason ||= "Login Failed"` re-assigns the local variable needlessly. How about `@reason = reason || "Login Failed"`?